classes/war/WarManager.php

```diff
@@ -147,20 +157,76 @@ public function beginOperation(int $operation_type, int $target_id, Patrol $patr
         }
     }
 
+    /**
+     * @throws RuntimeException
+     */
+    public function checkOperationValid(Operation $operation): bool {
+        if ($this->user->battle_id > 0) {
+            return false;
+        }
+        if ($operation->type != Operation::OPERATION_LOOT) {
+            $target = $this->system->db->query("SELECT `region_locations`.*, `regions`.`village` FROM `region_locations`
+            INNER JOIN `regions` on `regions`.`region_id` = `region_locations`.`region_id`
+            WHERE `id` = {$operation->target_id} LIMIT 1");
+            if ($this->system->db->last_num_rows == 0) {
+                return false;
+            }
+            $target = $this->system->db->fetch($target);
+            // must be at target location
+            $target_location = new TravelCoords($target['x'], $target['y'], $target['map_id']);
+            if ($this->user->location->fetchString() != $target_location->fetchString()) {
+                return false;
+            }
+        }
+        switch ($operation->type) {
+            case Operation::OPERATION_INFILTRATE:
+                // must be neutral or at war
+                if ($this->user->village->relations[$target['village']]->relation_type != VillageRelation::RELATION_NEUTRAL && $this->user->village->relations[$target['village']]->relation_type != VillageRelation::RELATION_WAR) {
```

Comment:
Is there anywhere else we do this? This could be a good candidate for a helper function like `village->villageIsAlly($target['village'])` or `WarManager::villagesAreAllied($this->user->village, $target['village'])`

Reply:
We also do this elsewhere:
```
if ($this->user->village->village_id == $user['village_id']) {
                $user_alignment = 'Ally';
            }
            else {
                $alignment = $this->user->village->relations[$user['village_id']]->relation_type;
                switch ($alignment) {
                    case VillageRelation::RELATION_NEUTRAL:
                        $user_alignment = 'Neutral';
                        break;
                    case VillageRelation::RELATION_ALLIANCE:
                        $user_alignment = 'Ally';
                        break;
                    case VillageRelation::RELATION_WAR:
                        $user_alignment = 'Enemy';
                        break;
                }
            }
```